### ISSUE TITLE
Add `directed` param to `osmdata_sc` fn

### ISSUE THREAD

**Issue by mpadge** [ropensci/osmdata#152]
This will enable return of directed graphs for `osmdata` calls on street networks. See discussion on [this `silicate` issue](https://github.com/hypertidy/silicate/issues/81). @mdsumner can be renamed accordingly at any stage

**Comment by mpadge** [comment]
Ah shite, OSM allows distinct `oneway.bicycle` tags, so this can't be a simple binary flag. Have to default to `directed = FALSE`, but also allow both `directed = TRUE` which reads all `oneway` flags, or `directed = "bicycle"` (or anything else with specific OSM oneway flags), which reads only `oneway.bicycle` (or whatever) flags.

**Comment by mpadge** [comment]
Mike, this now allows this:
```
> q <- opq ("48149 germany") %>%
       add_osm_feature (key = "highway", value = "secondary")
> osmdata_xml (q, filename = "ms.xml")
> x1 <- osmdata_sc (q, doc = "ms.xml", directed = FALSE)
> x2 <- osmdata_sc (q, doc = "ms.xml", directed = TRUE)
> sapply (x1, nrow)
          object object_link_edge             edge           vertex             meta
            2314              874              874              856                1
> sapply (x2, nrow)
          object object_link_edge             edge           vertex             meta
            2314             1637              874              856                1
```
where the latter only duplicates the ways that are **NOT** designated as one-way, and so 1637 < 874 * 2 = 1748. The `$object_lnk_edge` table is sorted like this:
```
> x2$object_link_edge
# A tibble: 1,637 x 3
   edge_      object_  native_
   <chr>      <chr>    <lgl>
 1 2buii3ZljE 10097831 TRUE
 2 ZhKixjXlaK 10097831 TRUE
 3 Njq641L9jf 10097831 TRUE
 4 OORg0wngIc 10097831 TRUE
 5 Sz5dgJhm7b 10097831 TRUE
 6 1Jhqo3rHk7 10097831 TRUE
 7 O5QMnmQwQr 10097831 TRUE
 8 2buii3ZljE 10097831 FALSE
 9 ZhKixjXlaK 10097831 FALSE
10 Njq641L9jf 10097831 FALSE
# ... with 1,627 more row
```
so all of the `native_ == TRUE` values for a given object are followed by the `native_ == FALSE` values for that object. Let me know if you see any more sensible ways to construct this, otherwise I'll consider it done.

**Comment by mdsumner** [comment]
Nice one, I'll have to let it sit for a bit to think about it. I definitely am more clear about what's going on now!  I see how `native_` is a bit odd for a layer intended as a transport network, but I actually like that it extends to this way of going about it. I'm a bit uncomfortable about duplicating like like, but I guess from your perspective this allows them to be stored one way or the other, or both. That could be achieved with a `undirected_` column, and you duplicate those [as here](https://github.com/hypertidy/silicate/issues/81#issuecomment-435833446). 


I guess that's what you were saying that I've finally clued into, use "directed_" rather than "native_" and if FALSE, then they must be duplicated for network use.   I need to dwell on it a bit.

**Comment by mpadge** [comment]
Yeah, you're right that this duplication of `native_` is a bit odd and quite likely counter to much of what `silicate` is trying to do. I'll modify back to original, non-duplicated form with an additional `undirected_` column (defaulting to `TRUE`). This is obviously better, coz instead of almost duplicating an entire `n x 3` table, it just adds another row, so only adds `n` items rather than `3n`. Much better. Thanks for clarification.

**Comment by mdsumner** [comment]
Well, still thinking out aloud - I'm wondering if the directedness property belongs on the `edge` table itself, rather than the `object_link_table` - because from the perspective of `SC.sf` these edge instances are "native" or not, they aren't directed in any special sense (because SF doesn't provide that concept, it doesn't provide any concepts at the part level, at least in theory). 

If an SF layer has two or more instances, then `native_` allows to keep a record of what that orientation *was*, not necessarily what it should or ought to be. It might be a self-intersecting reversal of the exact same edge - or it might be overlapping lines of various fragments, that run in a particular direction, or in both/either  directions. If we know that it's *not supposed to* be used for running .vx1 -> .vx0 then a `directed_` property on the `edge` table dictates that, and we then know that any non `native_` instance on the link table is some kind of rogue, which we ignore or fix as appropriate. 

Maybe that gives the right model? It's definitely a part of SC that's lacking right now.

**Comment by mdsumner** [comment]
Oh cool, missed your thumbs up!

One thing I need is minimal data examples,the simplest sets of edge and vertices to capture each needed scenario. But v glad you approve here.😎
